packages/schematics/angular/app-shell/index.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { normalize } from '@angular-devkit/core';
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  Tree,
+  chain,
+  schematic,
+} from '@angular-devkit/schematics';
+import 'rxjs/add/operator/merge';
+import * as ts from 'typescript';
+import {
+  addImportToModule,
+  addToNgModuleMetadata,
+  findNode,
+  getDecoratorMetadata,
+  getSourceNodes,
+  isImported,
+} from '../utility/ast-utils';
+import { InsertChange } from '../utility/change';
+import { AppConfig, getAppFromConfig, getConfig } from '../utility/config';
+import { getAppModulePath } from '../utility/ng-ast-utils';
+import { insertImport } from '../utility/route-utils';
+import { Schema as AppShellOptions } from './schema';
+
+
+// Helper functions. (possible refactors to utils)
+function getSourceFile(host: Tree, path: string): ts.SourceFile {
+  const buffer = host.read(path);
+  if (!buffer) {
+    throw new SchematicsException(`Could not find bootstrapped module.`);
+  }
+  const content = buffer.toString();
+  const source = ts.createSourceFile(path, content, ts.ScriptTarget.Latest, true);
+
+  return source;
+}
+
+function getServerModulePath(host: Tree, app: AppConfig): string | null {
+  const mainPath = `/${app.root}/${app.main}`;
+  const mainSource = getSourceFile(host, mainPath);
+  const allNodes = getSourceNodes(mainSource);
+  const expNode = allNodes.filter(node => node.kind === ts.SyntaxKind.ExportDeclaration)[0];
+  if (!expNode) {
+    return null;
+  }
+  const relativePath = <ts.StringLiteral> (<ts.ExportDeclaration> expNode).moduleSpecifier;
+  const modulePath = normalize(`/${app.root}/${relativePath.text}.ts`);
+
+  return modulePath;
+}
+// end helper functions.
+
+function addUniversalApp(options: AppShellOptions): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    // Copy options.
+    const universalOptions = {
+      ...options,
+      name: options.universalApp,
```

Comment:
Should check for `universalApp` option and throw or set default?

Reply:
`universalApp` is required per the app-shell schematic. (may have been added after your review)